### PR TITLE
TableScan filters to PyArrow DNF format

### PR DESCRIPTION
**Pull request by jdye64**
This PR adds the support to get the filters for a `LogicalPlan::TableScan` instance in the DNF format that is expected by PyArrow for performing predicate pushdown. Note this PR only adds support for `Expr::InList` for now.

An example of calling this functionality from Python goes like this

```python
# Rust table_scan instance handle obtained from LogicalPlan instance, "rel"
table_scan = rel.table_scan()
dnf_filters = table_scan.getDNFFilters()

# dnf_filters is of type `PyFilteredResult` from Rust. Contains the filters that are not supported by PyArrow DNF
# and also ones that can.
none_filterable_exprs = dnf_filters.io_unfilterable_exprs()
filterable_exprs = dnf_filters.filtered_exprs() # Tuple of (str, str, [str]) Ex: ('id', 'in', ['Int32(1)', 'Int32(2)'])

```

### CODE REVIEW

File: dask_planner/src/sql/logical/table_scan.rs
Comment (ayushdg) [inline dask_planner/src/sql/logical/table_scan.rs:50]: Does it make sense to check here if `expr` is a column or some scalar kind of type?
Can't think of a example but for cases where the expr instance maps to a binaryexpr or something that can't be applied directly, we'd want to use the unfiltered case.

File: dask_planner/src/sql/logical/table_scan.rs
Comment (sarahyurick) [inline dask_planner/src/sql/logical/table_scan.rs:39]: I like this for now. In https://github.com/dask-contrib/dask-sql/pull/1102 it also only deals with columns and literal values at the moment.

In the future we can add handling for more complex expressions. I have this listed in points 2 and 3 of https://github.com/dask-contrib/dask-sql/issues/1121 for DPP as well, so maybe down the road this PR and DPP can share some of the same logic to achieve this.

File: dask_planner/src/sql/logical/table_scan.rs
Comment (sarahyurick) [inline dask_planner/src/sql/logical/table_scan.rs:192]: Just wanted to ask why an `Expr::Placeholder` is used here instead of an `Expr::InList`?

**Review by sarahyurick — COMMENTED** [review]
Thanks @jdye64 ! Left a few comments, changes generally look good to me.

File: dask_planner/src/sql/logical/table_scan.rs
Comment (jdye64) [reply dask_planner/src/sql/logical/table_scan.rs:192]: Short answer: Laziness lol

Long answer: `PyExpr` currently requires a `LogicalPlan` instance to perform schema based queries. However, that is removed in #1085 so once that merges this snippet of code will cause a compilation error and need removed anyway. So knowing that I got a little lazy and just copied a `LogicalPlan` node from another test in the codebase.

You are right though, I should, and will update it to use `Expr::InList`

**Review by sarahyurick — APPROVED** [review]
LGTM

**Review by ayushdg — COMMENTED** [review]
Playing around with the example a bit, we end up getting `filterable_exprs` that looks something like: 
```
('df.a', 'in', ['Int64(1)', 'Int64(2)', 'Int64(3)', 'Int64(4)', 'Int64(5)'])
```

when passing this down to the reader we'd see the following issues:
`df.a` the reader doesn't understand this and is probably expecting `a`. I can just split by `.` on the python side but was curious if there's something better than `canonical` name that would give the column name.

`'Int64(1)'` Doesn't map to anything. Is it possible to return the literal values in this case? If not we would probably need to handle this on the python side.

**Comment by jdye64** [comment]
I need some pieces from #1085 to complete this so waiting on that to merge first.

File: dask_planner/src/sql/logical/table_scan.rs
Comment (ayushdg) [inline dask_planner/src/sql/logical/table_scan.rs:91]: For an unsupported scalar type we should raise an error and have it go via the unfiltered_exprs path

File: dask_planner/src/sql/logical/table_scan.rs
Comment (ayushdg) [inline dask_planner/src/sql/logical/table_scan.rs:97]: Would prefer the same check for `expr` being a alias or column type.

**Review by ayushdg — CHANGES_REQUESTED** [review]
Might also be worth adding a few tests on the rust side to ensure things are working as expected.

**Comment by jdye64** [comment]
@ayushdg had Rust tests before. Would have loved to have kept it but now that we using the GIL in these methods it seems there is a bug in PyO3 that is preventing me for doing so. Once again this issue only occurs when using the Python GIL in Rust tests. Take a look at this FAQ page. I tried all of their recommendations but none of them worked. Still kept hitting linker errors so the project fails to build if I include them.  https://pyo3.rs/v0.18.3/faq

Long story short I don't see a way we can add those Rust tests until that issue is resolved on the PyO3 side.

File: dask_planner/src/sql/logical/table_scan.rs
Comment (ayushdg) [inline dask_planner/src/sql/logical/table_scan.rs:141]: Just leaving this in as a note for followup work:
The approach of appending to the filtered list of expr's wouldn't return a dnf for more complex cases and right now is only limited to expr's that are `and` with each other.
eg: `a in (1,2,3,4,5) or b=5` would not get expanded by this logic.

**Review by ayushdg — APPROVED** [review]

**Pull request merged**
